src/flows/pipes/RawFluxEditor/FunctionsList/DynamicFunctionsList.tsx

```diff
@@ -26,18 +26,36 @@ const sortFuncs = (a, b) => {
 }
 
 const DynamicFunctionsList: FC<Props> = ({onSelect}) => {
+  const [eventSearchTerm, setEventSearchTerm] = useState('')
+  const [termRecorded, setTermRecorded] = useState('')
+  const [tooltipPopup, setTooltipPopup] = useState(false)
   const dispatch = useDispatch()
   const fluxFunctions = useSelector(getAllFluxFunctions)
+
   useEffect(() => {
     if (fluxFunctions.length === 0) {
       dispatch(getFluxPackages())
     }
   }, [])
 
-  const handleSelectItem = useCallback((func: FluxFunction) => {
-    onSelect(func)
-    event('Inject FluxDoc function into Flux Script')
-  }, [])
+  useEffect(() => {
+    if (tooltipPopup && eventSearchTerm !== termRecorded) {
+      event('flux.function.searched', {searchTerm: eventSearchTerm})
+      setTermRecorded(eventSearchTerm)
+    }
+    setTooltipPopup(false)
+  }, [eventSearchTerm, tooltipPopup])
+
```

Comment:
I set up a state to keep track of tooltip pop up status. when tooltip component renders, tooltip pop up state gets set to true. When true, we check if search term is a new search term we haven't recorded before, if so, we can the search value to event metrics. Keeping termRecorded state is necessary because, it ensures search term will be sent to metrics once.  that is, until we have a new search term

Reply:
@TCL735 what are your thoughts on this?

Reply:
If we like this approach, I will add this to data explorer flux help panel